### ISSUE TITLE
QueryAsync - Does it work correctly?

### ISSUE THREAD

**Issue by LyalinDotCom** [praeclarum/sqlite-net#120]
I am wondering if QueryAsync is working as expected, I am trying to get it to work with a simple Twitter test app, im saving posts to the database

If I run the code below as shown its not working, it seems that the question mark is not being replaced with the parameter im passing in

Code:

```
    var sqlcon = SQLiteDbManager.SqlAsyncConnection;
            var searchResults = await sqlcon.QueryAsync<SavedTwitterPost>
                ("SELECT * FROM SavedTwitterPost WHERE text LIKE '%?%'", SearchWord);
```

But hardcoding the same string instead of ? placeholder does work and returns data. Also testing using a question mark in a straight equals scenario works. For example "WHERE something = ?" works, but not the LIKE command example above.

Also maybe im just creating the wrong SQLite query all together :(. Can anyone confirm if they have this working? for now ill just use String.Format to insert my query word for Like statement.

**Comment by golddragon007** [comment]
Yes, the LIKE operator works correctly, just you use it at wrong way. Here is how you need to use it:

    var sqlcon = SQLiteDbManager.SqlAsyncConnection;
            var searchResults = await sqlcon.QueryAsync<SavedTwitterPost>
                ("SELECT * FROM SavedTwitterPost WHERE text LIKE ?", "%".SearchWord."%");

This works a lot of other languages/implementations too, not only in this one. You need to concat the % with your param.

**Comment by skartknet** [comment]
wow, an answer 4 years later, and the ticket is still open ... XD. Yes, the reply is correct and it works.

**Comment by jamesryanbell** [comment]
@praeclarum looks like this can be closed, thanks @golddragon007
